Svc/BufferManager/BufferManager.cpp

```diff
@@ -270,10 +270,31 @@ namespace Svc {
 
   }
 
-  // ----------------------------------------------------------------------
+  U32 BufferManager :: getBufferContext(const U32 managerId, const U32 bufferId) {
+      // Context is a concatenation of IDs, so check for overflows
+      FW_ASSERT(managerId < 0xFFFF, managerId);
+      FW_ASSERT(bufferId <= 0xFFFF, bufferId);
+
+      U32 context = (managerId << 16) | bufferId;
+      FW_ASSERT(context != Fw::Buffer::NO_CONTEXT); // Ensure we didn't collide with not context
+      return context;
+  }
+
+  U32 BufferManager :: getManagerIdFromBufferContext(const U32 context) {
+      FW_ASSERT(context != Fw::Buffer::NO_CONTEXT); // Ensure we didn't receive with not context
+      return (context >> 16) & 0x7FFF;
```

Comment:
Why `0x7FFF`? Aren't you storing manager as a 16 bit number

Reply:
0xFFFFFFFF is an invalid context. Thus this code forces the top bit to be 0.

Reply:
That makes sense, you may want to assert above in `getBufferContext` that manager ID is less than `0x7FFF`, rather than `0xFFFF` to avoid confusion, since I saw the mismatch as a red flag because the bounds checks were different.